apps/evm/lib/evm/vm.ex

```diff
@@ -144,4 +144,41 @@ defmodule EVM.VM do
 
     {final_machine_state, n_sub_state, n_exec_env}
   end
+
+  def log(operation, machine_state) do
```

Comment:
There seems to be a lot of stuff going on in this function. Do you think it might make sense to create a debugging module that allows us to call individual functions? Not sure if it's overkill, but just a suggestion.

Reply:
Good call!

I extracted this [out into it's own module](https://github.com/poanetwork/mana/pull/164/commits/28f4533da5a3c83fb015fc4ad2750c612fa369b3).